client/features/settings-page/Settings.tsx

```diff
@@ -1,9 +1,118 @@
-import React from 'react';
-import useDocumentTitle from '@hooks/useDocumentTitle';
+import React, { useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { Box, Typography, TextField, Button } from '@mui/material';
+import styles from '@styles/SettingsPage.module.css';
+import {
+  useFetchUserProfile,
+  useUpdateUserProfile,
+} from '@hooks/useUserCrudOperations';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { userDisplayNameSchema } from '@validators/validateUserDisplayName';
+import { useQueryClient } from '@tanstack/react-query';
+import useDebounce from '@hooks/useDebounce';
+import { AxiosError } from 'axios';
+import { useNotification } from '@hooks/useNotification';
+import Loader from '@components/common/Loader';
 
-const Settings = () => {
-  useDocumentTitle('Settings');
-  return <div>Settings Placeholder</div>;
+interface UserProfileFormData {
+  name: string;
+}
+
+const Settings: React.FC = () => {
+  const { showNotification } = useNotification();
+
+  const queryClient = useQueryClient();
+  const { data: userProfile, isFetching } =
+    useFetchUserProfile('Settings Page');
+
+  const updateUserProfile = useUpdateUserProfile('Settings Page');
+
+  const {
+    control,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<UserProfileFormData>({
+    resolver: zodResolver(userDisplayNameSchema),
+    defaultValues: {
+      name: userProfile?.name,
+    },
+  });
+
+  useEffect(() => {
+    if (userProfile) reset({ name: userProfile.name });
+  }, [userProfile, reset]);
+
+  const onSubmit = async (data: UserProfileFormData) => {
+    try {
+      await updateUserProfile.mutateAsync(data);
+
+      await queryClient.invalidateQueries({ queryKey: ['getUserProfile'] });
+
+      showNotification('Display name updated successfully', 'success');
+      reset();
```

Comment:
here because you're invalidating the query (and thus forcing a re-fetch of new values) and then immdiately after resetting the form state to its default value of `userProfile.name` which is just the existing (previous) name, so this call to reset() might not be necessary (there shouldn't be any actual visual change in the UI)

The only thing I could think of that would make reset() necessary is if, for example we wanted to track this form change somehow, which it looks like we're not doing anywhere